user_guide_src/source/libraries/validation/030.php

```diff
@@ -1,7 +1,7 @@
-<div class="alert alert-danger" role="alert">
-    <ul>
-    <?php foreach ($errors as $error): ?>
-        <li><?= esc($error) ?></li>
+<?php if (! empty($errors)): ?>
+    <div class="alert alert-danger">
+    <?php foreach ($errors as $field => $error): ?>
+        <p><?= esc($error) ?></p>
```

Comment:
By the way, why do you want to change to `<p>` instead of `<li>`?
Or why do you want it to be the same code as Model's?

Reply:
> Or why do you want it to be the same code as Model's?

for consistency, it's actually the same twice.
But you can leave it as is, with ul li. Will just put `if (! empty($errors))` around it

Reply:
I don't see that we need consistency in this case.

This section is for *Customizing* Error Display.
https://codeigniter4.github.io/CodeIgniter4/libraries/validation.html#customizing-error-display

The sample in Model is for In-Model Validation.
https://codeigniter4.github.io/CodeIgniter4/models/model.html#in-model-validation